modules/renter/hostdb/scan.go

```diff
@@ -142,7 +142,7 @@ func (hdb *HostDB) threadedProbeHosts() {
 
 			// If 'maxActiveHosts' has not been reached, add the host to the
 			// activeHosts tree.
-			if _, exists := hdb.activeHosts[hostEntry.NetAddress]; !exists && len(hdb.activeHosts) < maxActiveHosts {
+			if _, exists := hdb.activeHosts[hostEntry.NetAddress]; exists || len(hdb.activeHosts) < maxActiveHosts {
```

Comment:
This guy was causing the negative currency errors and host tree restructuring. Lack of safety on the hostdb tree can be blamed for this.

Adding safety would be a pretty major refactor.